src/screens/HomeScreen.jsx

```diff
@@ -1,18 +1,52 @@
-import React, { useContext } from 'react';
-import { Text, View } from 'react-native';
-import LogInOrOutButton from '../components/LogInOrOutButton';
-import UserContext from '../contexts/UserContext';
+import React, { useEffect, useState } from 'react';
+import { View, ScrollView } from 'react-native';
+import HeaderComponent from '../components/HeaderComponent';
+import ListingView from '../components/ListingView';
+import ListingHeader from '../components/ListingHeader';
+import { getListings } from '../api';
 
 const HomeScreen = () => {
-    const { user } = useContext(UserContext);
+    const [tickets, setTickets] = useState([]);
+    const [textbooks, setTextBooks] = useState([]);
+    const [other, setOther] = useState([]);
+
+    useEffect(() => {
+        try {
+            const temp = getListings();
+            temp.then(value => {
+                for (let i = 0; i < value['listings'].length; i++) {
+                    if (value['listings'][i]['type'] === "ticket") {
+                        setTickets((tickets) => [...tickets, value['listings'][i]])
+                    } if (value['listings'][i]['type'] === "textbook") {
+                        setTextBooks((textbooks) => [...textbooks, value['listings'][i]])
+                    }
+                    if (value['listings'][i]['type'] === "other") {
+                        setOther((other) => [...other, value['listings'][i]])
+                    }
+                }
+            });
+        } catch {
+            console.log('errror :(');
+        }
+    }, []);
 
     return (
         <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
-            <Text>
-                {user ? `Logged in as ${user.firstName} ${user.lastName}` :
-                    "Log in with your UMich Google account:"}
-            </Text>
-            <LogInOrOutButton />
+            <ScrollView>
+                <HeaderComponent />
+                <View >
+
+                    <ListingHeader category={tickets.length != 0 ? `${tickets[0]['type']}` : ""} />
```

Comment:
It would probably be cleaner to conditionally render this whole block. Something like:

```suggestion
                    {tickets.length > 0 && (
                        <>
                            <ListingHeader category="Tickets" />
                            <ListingView list={tickets} />
                        </>
                     )}
```

Same for the other three categories